src/catalog/database_catalog.cpp

```diff
@@ -1330,6 +1330,38 @@ std::vector<std::pair<common::ManagedPointer<storage::index::Index>, const Index
   return index_objects;
 }
 
+bool DatabaseCatalog::CopyPR(common::ManagedPointer<transaction::TransactionContext> txn,
+                             storage::ProjectedRow *table_pr, storage::ProjectedRow *index_pr, index_oid_t index_oid,
+                             common::ManagedPointer<terrier::storage::SqlTable> table,
+                             storage::TupleSlot table_tuple_slot) {
+  const auto key_schema = GetIndexSchema(txn, index_oid);
+  const auto index = GetIndex(txn, index_oid);
```

Comment:
This is an expensive operation to perform for each tuple. We should stash this in the `StorageInterface` at instantiation. Also, didn't you already have this in the calling scope?